inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/SortSourceServiceImpl.java

```diff
@@ -296,14 +305,26 @@ private Map<String, CacheZone> parseCacheZones(
 
         // group them by cluster tag.
         Map<String, List<SortSourceStreamSinkInfo>> tag2SinkInfos = sinkInfoList.stream()
+                .filter(sink -> {
+                    if (StringUtils.isBlank(sortClusterTag)) {
+                        return true;
+                    }
+                    return groupInfos.get(sink.getGroupId()).getClusterTag().equals(sortClusterTag);
```

Comment:
Maybe there will be NPE?

Reply:
The NPE will be thrown when there is a stream sink of a groupId  but this groupId does not exist in inlong_group table. I will add one more filter condition to avoid this case.